dismantle-arm-xml/src/Dismantle/ARM/ASL.hs

```diff
@@ -93,10 +97,19 @@ data ASLContext = ASLContext { ctxASLEncoding :: Maybe (ASL.Instruction, ASL.Ins
                              , ctxSourceLoc :: Maybe SrcLoc
                              }
 
-data ASLState = ASLState { stMaskTrie :: ARMMaskTrie [(String, ARMBitMask BM.QuasiBit)]
-                         , stEncodingMap :: Map String (ASL.Instruction, ASL.InstructionEncoding)
+data SizedTrie n =
+  SizedTrie { _stRepr :: NR.NatRepr n
+            , _stTrie :: BM.MaskTrie BM.QuasiBit n [(String, BM.BitMask n BM.QuasiBit)]
+            }
+
+$(L.makeLensesFor [("_stTrie", "stTrie")] ''SizedTrie)
+
+data ASLState = ASLState { _stMaskTrie :: MapF.MapF NR.NatRepr SizedTrie
+                         , _stEncodingMap :: Map String (ASL.Instruction, ASL.InstructionEncoding)
```

Comment:
This is a big PR, but I suspect this change (and the removal of the `type ARMMaskTrie a` above) are the core changes that pretty much drive everything else.

Also, I like the `makeLensesFor` trick, I hadn't known about that one.